### DISCUSSION TITLE
Regarding "Hardcoded password".

### DISCUSSION THREAD

**Discussion by mightysai1997** [zricethezav/gitleaks#1171] [Q&A]
whether Gitleaks will detect "Hardcoded password" or not.

I added 

var password = “123password123”
 
to my nodejs file.  The NodeJS file change triggered a NodeJS error (not secrets).

why it is not detecting ?

**Comment by rgmz** [comment]
It's possible that secret is being detected but ignored because "password123" is a common test/sample password.

Try changing it to a more realistic password and see if that gets detected.

**Reply by sharovik** [reply]
I would say, the most common cases are the usage of weak passwords. I believe we need to add the rule for that. 

What if we will use something like this in the gitleaks config?
```yaml
[[rules]]

# Unique identifier for this rule
id = "password"

# Short human readable description of the rule.
description = "password"

# Golang regular expression used to detect secrets. Note Golang's regex engine
# does not support lookaheads.
regex = '''(?im)(password)+.+(['"].*['"])'''

# Array of strings used for metadata and reporting purposes.
tags = ["password",]

# Int used to extract secret from regex match and used as the group that will have
# its entropy checked if `entropy` is set.
secretGroup = 2

# Float representing the minimum shannon entropy a regex group must have to be considered a secret.
entropy = 3.5

# Keywords are used for pre-regex check filtering. Rules that contain
# keywords will perform a quick string compare check to make sure the
# keyword(s) are in the content being scanned. Ideally these values should
# either be part of the idenitifer or unique strings specific to the rule's regex
# (introduced in v8.6.0)
keywords = [
  "auth",
  "password",
  "token",
]
```

Just create the `gitleaks.toml`  file and add this rule into this file. Then, run the gitleaks command with this config
`gitleaks detect -v -c gitleaks.toml`

**Comment by mightysai1997** [comment]
I have added the 
var password = "S@LtPepp3r@";
in the index.js file and I committed it.

Even though  Git leaks was not detecting that hard coded password as vulnerability. 

let me known if any thing that needs to add.

**Comment by mightysai1997** [comment]
Hi Richard Gomez,
 Is there any update on the above mention comment.

I want to know whether the gitleaks  was pulling latest commit or not.

**Comment by mightysai1997** [comment]
base) chandra@chandra-Aspire-A715-42G:~/pygoat$ gitleaks detect -v -c gitleaks.toml
INFO[0000] opening .                                    
INFO[0000] scan time: 19 milliseconds 663 microseconds  
INFO[0000] No leaks found

**Reply by sharovik** [reply]
As I can see, you have in your string the different quotes, which are not valid, at least in javascript.
![image](https://github.com/gitleaks/gitleaks/assets/3922963/15743741-7f6b-43ad-8e84-dfca6205e2dc)

So, please make sure:
1. update your code to use the right quotes
`var password = "123password123"` instead `var password = “123password123”`
2. update regexp to have your quotes `(password)+.+(['"“].*['"”])`, if you want to keep them

**Comment by mightysai1997** [comment]
Issue get resolved
